test/Dialect/FIRRTL/lower-types.mlir

```diff
@@ -663,3 +663,18 @@ firrtl.circuit "AnnotationsRegOp" {
   // CHECK: firrtl.regreset
   // CHECK-SAME: annotations = [{b = "b"}]
 }
+
+
+// -----
+
+// Test that subfield annotations on wire are lowred to appropriate instance based on target.
+firrtl.circuit "AnnotationsBundle" {
+  firrtl.module @AnnotationsBundle() {
+    %bar = firrtl.wire  {annotations = [{one, target = [".qux"]}, {target = ["[1]", ".baz"], two}]} : !firrtl.vector<bundle<baz: uint<1>, qux: uint<1>>, 2>
+
+      // CHECK: %bar_0_baz = firrtl.wire  : !firrtl.uint<1>
+      // CHECK: %bar_0_qux = firrtl.wire  {annotations = [{one}]} : !firrtl.uint<1>
+      // CHECK: %bar_1_baz = firrtl.wire  {annotations = [{two}]} : !firrtl.uint<1>
+      // CHECK: %bar_1_qux = firrtl.wire  {annotations = [{one}]} : !firrtl.uint<1>
```

Comment:
Actually, this may be wrong for `.qux`. This is acting like a wildcard, when it actually shouldn't be applied. The target here is `~AnnotationsBundle|AnnotationBundle>bar.qux`. That should only apply to `bar.qux` not anything which matches `qux`.

Reply:
Okay, so I was considering the case, when we want to apply the annotation to all the elements of ``bar[ ].qux``
So, the suffix should match exactly, no wild card matching, and in the above example it should be ignored ?
Another example is, if I specify the annotation on, ``a.b``, and ``b`` has subfields, ``a.b.c`` and ``a.b.d``, the annotation would be ignored ?

Reply:
Currently, the only way to match `bar[*].qux` would be to enumerate `bar[0].qux`, `bar[1].qux`, `...`.

In the above example, the `one` annotation should be dropped. We can add in checking of this during parsing in the future.

For that example, I'd change it to `target = ["[0]"]` to test the path of correctly mapping to multiple lowered wires.

An annotation on `a.b` should get mapped to `a.b.c` and `a.b.d`.

@jackkoenig: does that sound okay to you?

Reply:
> @jackkoenig: does that sound okay to you?

I'm a bit late here but yeah seems reasonable to me